pkg/virt-controller/watch/migration/migration.go

```diff
@@ -1334,8 +1347,17 @@ func (c *MigrationController) sync(key string, migration *virtv1.VirtualMachineI
 			!vmi.Status.MigrationState.Failed &&
 			!vmi.Status.MigrationState.Completed {
 
-			return c.handleMarkMigrationFailedOnVMI(migration, vmi)
+			err = c.handleMarkMigrationFailedOnVMI(migration, vmi)
+			if err != nil {
+				return err
+			}
 		}
+
+		if migration.Status.Phase != virtv1.MigrationFailed {
+			return nil
+		}
+
+		return descheduler.RemoveDeschedulerAnnotationFromSourcePod(c.clientset, migration, c.podIndexer)
```

Comment:
Shouldn't we remove the annotation only when the migration completes? Otherwise the descheduler might try to evict the source pod while is transferring the VM no?

Reply:
If we reach this code it means that the migration fails, and `Failed` is a final status for migrations.
We add the annotation at the beginning and the source pod is annotated for the entire migration process.
In case the migration succeeds the source pod is completed and can remain annotated. In case 
the migration fails we drop the annotation.
IOW: this is under a switch case were the migration is `virtv1.MigrationPreparingTarget, virtv1.MigrationTargetReady, virtv1.MigrationFailed` and we early exit in case of:
```
if migration.Status.Phase != virtv1.MigrationFailed {
	return nil
}
```

Reply:
Sorry somehow the switch case tricked me, all good.